### PR TITLE
[VarExporter] Add `Hydrator` section

### LINKED ISSUE #16827: [VarExporter] Add Hydrator::hydrate() and preserve PHP referenc…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46452
| PR author(s) | @nicolas-grekas
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/16827

### CODE REVIEW

File: components/var_exporter.rst [hunk 1]
```diff
@@ -95,30 +95,39 @@ file looks like this::
...
-Instantiating PHP Classes
--------------------------
+Instantiating & Hydrating PHP Classes
+-------------------------------------
 
-The other main feature provided by this component is an instantiator which can
-create objects and set their properties without calling their constructors or
-any other methods::
+Istantiator
+~~~~~~~~~~~
```
Comment (alamirault) [inline components/var_exporter.rst]:
```suggestion
Instantiator
~~~~~~~~~~~~
```

Comment (alexandre-daubois) [reply components/var_exporter.rst]: Oops, thanks! That's updated

**Comment by javiereguiluz** [comment]
Alex, thanks a lot for this contribution. The VarExporter component doesn't receive many doc PRs ... so we appreciate them a lot!

While merging we did some tweaks (see https://github.com/symfony/symfony-docs/commit/013b6d1233adb8da6d5f4c55d1636a23416b9e15):

* Adding the `.. _instantiating-php-classes:` reference is needed because we changed the headline, so the anchor link will change. This will "break" direct links to that section: `https://symfony.com/doc/current/components/var_exporter.html#instantiating-php-classes` (that's why we always add a hidden reference to the previous headline contents when we change them)
* I reversed some changes in the comments that capitalized the first letter: `// Creates ...` vs `// creates ...` Of course your proposal is not wrong ... but we mostly use the second option because comments are usually one line and the capital letter looks unbalanced with the following code, which usually starts with a lowercase letter or the `$` variable indicator. In the future we can discuss about changing this 🙏 

Thanks!

**Comment by alexandre-daubois** [comment]
Amazing, thanks @javiereguiluz!

**Pull request merged**